core/src/test/java/org/infinispan/xsite/irac/IracTombstoneUnitTest.java

```diff
@@ -117,13 +126,24 @@ private static IracMetadata createIracMetadata() {
       return Mockito.mock(IracMetadata.class);
    }
 
+   private static BlockingManager createBlockingManager() {
+      BlockingManager blockingManager = Mockito.mock(BlockingManager.class);
+      Mockito.when(blockingManager.asExecutor(ArgumentMatchers.anyString())).thenReturn(new WithinThreadExecutor());
+      Mockito.when(blockingManager.thenComposeBlocking(ArgumentMatchers.any(), ArgumentMatchers.any(), ArgumentMatchers.any())).thenAnswer(invocationOnMock -> {
+         CompletionStage<Void> stage = invocationOnMock.getArgument(0);
+         Function<Void, CompletionStage<Object>> f = invocationOnMock.getArgument(1);
+         return stage.thenComposeAsync(f);
```

Comment:
Did you want to just use `thenCompose`? Otherwise this will run on the common thread pool.

Reply:
ops. fixed!